heat/core/dndarray.py

```diff
@@ -653,6 +653,84 @@ def fill_diagonal(self, value: float) -> DNDarray:
 
         return self
 
+    def __process_key(arr: DNDarray, key: Union[int, Tuple[int, ...], List[int, ...]]) -> Tuple:
+        """
+        Private method for processing keys for indexing. Returns wether advanced indexing is used as well as a processed key and self.
+        A processed key:
+        - doesn't cotain any ellipses or newaxis
+        - all Iterables are converted to ``DNDarrays``
+        - has the same dimensionality as the ``DNDarray`` it indexes
+
+        Parameters
+        ----------
+        key : int, slice, Tuple[int,...], List[int,...]
+            Indices for the tensor.
+        """
+        advanced_indexing = False
+        if isinstance(key, DNDarray):
+            # DNDARRAY CURRENTLY DOES NOT IMPLEMENT the Iterable interface, need to define __iter__()
```

Comment:
We could implement the Iterable interface for non-distributed DNDarrays, but we need to give some thought to what to fall back on when `split is not None`.

Reply:
Well, `iter(...)` is just an Iterator over the first dimension, right? So I think
"""python
def __iter__(self):
    if self.split = 0:
        warnings.warn("Iterating over the split dimension is strongly discouraged")
    for i in range(len(self)):
       yield self[i]
"""
would work as expected